lstchain/datachecks/containers.py

```diff
@@ -296,40 +356,40 @@ def fill_pixel_wise_info(self, table, mask, histogram_binnings,
+                                       for row in time])
             # mean and std dev for each pixel through the whole subrun:
-            self.time_mean = np.nanmean(selected_entries, axis=0)
-            self.time_stddev = np.nanstd(selected_entries, axis=0)
+            self.time_mean = np.nanmean(time, axis=0)
+            self.time_stddev = np.nanstd(time, axis=0)
             # Now the average time in the camera, for each event:
-            tmean = np.nanmean(selected_entries, axis=1)
-
-            # tile it to the same shape as time, to allow subtracting it from each
-            # pixel's pulse time:
-            camera_time_mean = np.tile(tmean, (time.shape[1], 1)).transpose()
-            # from camera mean time, to the same but excluding one pixel at a
-            # time:
-            camera_valid_pixels = np.tile(n_valid_pixels,
-                                          (time.shape[1], 1)).transpose()
-            rest_of_camera_valid_pixels = camera_valid_pixels - \
-                                          np.ones(camera_valid_pixels.shape)
-            mean_t_of_rest_of_camera = ((camera_time_mean *
-                                         camera_valid_pixels - time) /
-                                        rest_of_camera_valid_pixels)
-
-            relative_time_t = time - mean_t_of_rest_of_camera
-            selected_entries = np.where(charge > 1, relative_time_t, np.nan)
-            self.relative_time_mean = np.nanmean(selected_entries, axis=0)
-            self.relative_time_stddev = np.nanstd(selected_entries, axis=0)
+            tmean = np.nanmean(time, axis=1)
+
+            # We do the calculation of the relative times event by event,
+            # instead of using events*pixels matrices, because keeping all
+            # necessary matrices in memory to do it in one go results in too
+            # large memory use (>5GB)
+            for ievt, event_pixtimes in enumerate(time):
+                # for each pixel we want the mean time of all the other pixels:
+                mean_t_other = np.ones_like(event_pixtimes) * tmean[ievt]
+                mean_t_other *= n_valid_pixels[ievt]
+                mean_t_other -= event_pixtimes
+                mean_t_other /= (n_valid_pixels[ievt] - 1)
+                time[ievt] -= mean_t_other
+
+            # Now time contains the times of each pixel relative to the averag
+            # of the rest of the pixels in the same event
+
+            self.relative_time_mean = np.nanmean(time, axis=0)
+            self.relative_time_stddev = np.nanstd(time, axis=0)
 
             if event_type == 'flatfield':
                 return
 
-            selected_entries = np.where(charge > 30, time, np.nan)
-            self.time_mean_above_030_pe = np.nanmean(selected_entries, axis=0)
-            self.time_stddev_above_030_pe = np.nanstd(selected_entries, axis=0)
+            time = np.where(charge > 30, time, np.nan)
```

Comment:
Using masked arrays with `copy=False` should be better in terms of memory usage and speed:

```
masked_peak_time = np.ma.array(peak_time, mask=charge < 30, copy=False)
self.time_mean_above_030_pe = masked_peak_time.mean(axis=0).filled(np.nan)
self.time_stddev_above_030_pe = masked_peak_time.std(axis=0).filled(np.nan)
```